bsuite/utils/gym_wrapper.py

```diff
@@ -93,3 +94,77 @@ def reward_range(self) -> Tuple[float, float]:
   def __getattr__(self, attr):
     """Delegate attribute access to underlying environment."""
     return getattr(self._env, attr)
+
+
+class ReverseGymWrapper(dm_env.Environment):
+    """A wrapper that converts an OpenAI gym environment to a dm_env.Environment"""
+
+    def __init__(self, gym_env:gym.Env):
+        self.gym_env = gym_env
+        # convert gym action and observation spaces to dm_env specs
+        self._observation_spec = self.space2spec(self.gym_env.observation_space, "observations")
+        self._action_spec = self.space2spec(self.gym_env.action_space, "actions")
+
+    def reset(self):
+        self.gym_env.reset()
+
+    def step(self, action):
+        """convert gym step result (observations, reward, done, info) to dm_env TimeStep"""
+        gym_step_res = self.gym_env.step(action)
+        _obs = gym_step_res[0]
+        _reward = gym_step_res[1]
+        _done = gym_step_res[2]
+
+        if _done:
+            return dm_env.TimeStep(dm_env.StepType.LAST, _reward, None, _obs)
+        else:
+            return dm_env.TimeStep(dm_env.StepType.MID, _reward, None, _obs)
+
+    def close(self):
+        self.gym_env.close()
+
+    def observation_spec(self):
+        return self._observation_spec
+
+    def action_spec(self):
+        return self._action_spec
+
+    def space2spec(self, space:gym.Space, name:str=None):
```

Comment:
Does this need to be a public method?

Reply:
It should also be a function, rather than a method.